hanami-utils.gemspec

```diff
@@ -18,6 +18,8 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.required_ruby_version = '>= 2.3.0'
 
+  spec.add_dependency 'transproc', '~> 1.0'
```

Comment:
@davydovanton I'm torn about this. Generally speaking, I'd like the idea of having no deps for `utils`.

On the other hand, because `Utils::Hash` (and `Utils::String`) are widely used by our other gems, then developers will need to take care to manually install `transproc` and to maintain the dependency level that we use.

Example: I'm using only `hanami-controller` and `hanami-router` in a project of mine. So my `Gemfile` looks like this:

```ruby
# ...
gem 'hanami-controller', '~> 1.0'
gem 'hanami-router', '~> 1.0'
```

There is not even `hanami-utils` in my `Gemfile`. It would be odd to add `transproc`, as soft-dependency for a dependency that I don't even list in my `Gemfile`.

An additional maintenance burden for us would be the version of `transproc` that devs will use. Let's get back to our example, I add the gem:

```ruby
# ...
gem 'transproc'
```

We don't have any guarantee that a developer will pick the right version. That `Gemfile` instruction means "install the latest stable version of `transproc`", which may not match what we need in `utils`. Right now we enforce `transproc` 1.x series, but we can't prevent a developer to do `bundle update` and get 2.0.

Please note that `transproc` 2.0 isn't out, it was just an example.

---

Punchline is: it's probably better to add that direct dep in gemspec.